### PR TITLE
AG-11763 - new module validations

### PR DESCRIPTION
**Pull request by peterjrreynolds**

### CODE REVIEW

File: packages/ag-grid-community/src/validation/resolvableModuleNames.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    CheckboxCellRendererModule: ['AllCommunityModule'],
+    ClientSideRowModelHierarchyModule: ['RowGroupingModule', 'PivotModule', 'TreeDataModule'],
+    ColumnFilterModule: ALL_COLUMN_FILTERS,
+    ColumnGroupHeaderCompModule: ['AllCommunityModule'],
+    ColumnGroupModule: ['AllCommunityModule'],
+    ColumnHeaderCompModule: ['AllCommunityModule'],
+    ColumnMoveModule: ['AllCommunityModule'],
+    ColumnResizeModule: ['AllCommunityModule'],
+    CommunityCoreModule: ['AllCommunityModule'],
+    CsrmSsrmSharedApiModule: [],
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/validation/resolvableModuleNames.ts]: Won't this give odd output?

File: packages/ag-grid-community/src/validation/rules/gridOptionsValidations.ts [hunk 2]
```diff
@@ -369,6 +399,8 @@ const GRID_OPTION_VALIDATIONS: () => Validations<GridOptions> = () => {
         treeDataChildrenField: {
             module: 'SharedTreeDataModule',
         },
+        undoRedoCellEditing: { module: 'UndoRedoEditModule' },
+        valueCache: { module: 'ValidationModule' },
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/validation/rules/gridOptionsValidations.ts]: Is this correct?

File: packages/ag-grid-community/src/validation/resolvableModuleNames.ts (hunk 1 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/validation/resolvableModuleNames.ts]: Fixed

File: packages/ag-grid-community/src/validation/rules/gridOptionsValidations.ts (hunk 2 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/validation/rules/gridOptionsValidations.ts]: Fixed

**Pull request merged**